<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>DrugGCN &mdash; IMPROVE 0.1 documentation</title><link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> IMPROVE
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="whatis.html">What is the IMPROVE project?</a></li>
<li class="toctree-l1"><a class="reference internal" href="install.html">Installing IMPROVE</a></li>
<li class="toctree-l1"><a class="reference internal" href="models.html">Models</a></li>
<li class="toctree-l1"><a class="reference internal" href="curating_introduction.html">Introduction to Curating a Community Model</a></li>
<li class="toctree-l1"><a class="reference internal" href="curating.html">Curating a Community Model</a></li>
<li class="toctree-l1"><a class="reference internal" href="candle_1.html">Creating a CANDLE Benchmark</a></li>
<li class="toctree-l1"><a class="reference internal" href="candle_2.html">Loading Data Using the candle.file_utils Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="git-branch-tag-names.html">GitHub branch and tag names and definitions</a></li>
<li class="toctree-l1"><a class="reference internal" href="using-candle-data-dir.html">Tutorial CANDLE_DATA_DIR</a></li>
<li class="toctree-l1"><a class="reference internal" href="running-containers.html">How to run a container</a></li>
<li class="toctree-l1"><a class="reference internal" href="building-containers.html">How to build a container</a></li>
<li class="toctree-l1"><a class="reference internal" href="deploying-containers.html">How to deploy a container</a></li>
<li class="toctree-l1"><a class="reference internal" href="drp_overview.html">Drug Response Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="model_curation.html">Model Curation</a></li>
<li class="toctree-l1"><a class="reference internal" href="curated_models.html">Curated Models</a></li>
<li class="toctree-l1"><a class="reference internal" href="benchmarks.html">Benchmarks</a></li>
<li class="toctree-l1"><a class="reference internal" href="acknowledgment.html">Acknowledgments</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">IMPROVE</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a></li>
      <li class="breadcrumb-item active">DrugGCN</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/DrugGCN.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="section" id="druggcn">
<h1>DrugGCN<a class="headerlink" href="#druggcn" title="Permalink to this headline">¶</a></h1>
<p>DrugGCN is a framework for the prediction of <strong>Drug</strong> response using a Graph Convolutional Network (<strong>GCN</strong>).</p>
<div class="section" id="structure">
<h2>Structure<a class="headerlink" href="#structure" title="Permalink to this headline">¶</a></h2>
<p>DrugGCN uses a graph convolutional network model to learn genomic features of cell lines with a graph structure for drug response prediction. This framework integrates protein-protein interaction (PPI) network data and gene expression data utilizing genes with high predictive power to construct a model for each drug. The input of the GCN model is an undirected input graph G=(V,E,W) where V is a set of vertices, E is a set of edges, and W is a weighted adjacency matrix. Vertices and edges of the graph indicate genes and interactions between genes respectively. The use of localized filters in DrugGCN aids in the detection of local features in a biological network such as subnetworks of genes that can contribute together to drug response.</p>
</div>
<div class="section" id="data">
<h2>Data<a class="headerlink" href="#data" title="Permalink to this headline">¶</a></h2>
<div class="section" id="data-sources">
<h3>Data sources<a class="headerlink" href="#data-sources" title="Permalink to this headline">¶</a></h3>
<p>The primary data sources that have been used to construct datasets for model training and testing (i.e., ML data) include:
- GDSC - cell line and drug ids, treatment response, cell line omics data
- Library of Integrated Network-Based Cellular Signatures (LINCS) L1000 project - list of genes
- STRING - PPI network data</p>
</div>
<div class="section" id="raw-data">
<h3>Raw data<a class="headerlink" href="#raw-data" title="Permalink to this headline">¶</a></h3>
<p>The raw data is downloaded from GDSC website (version 6.0):</p>
<ol class="arabic simple">
<li><p>Dose-independent drug response values. <cite>PANCANCER_IC.csv</cite>: drug and cell ids, IC50 values and other metadata (223 drugs and 948 cell lines).</p></li>
<li><p>Cancer sample information. <cite>PANCANCER_Genetic_feature.csv</cite>: 735 binary features that include coding variants and copy number alterations.</p></li>
<li><p>Drug information. <cite>drug_smiles.csv</cite>: SMILES strings of drug molecules. The SMILES were retrieved from PubChem using CIDs (Druglist.csv). The script <cite>preprocess.py</cite> provides functions to generate this file.</p></li>
</ol>
<p>All these data types were provided with the GraphDRP repo and are available in <a class="reference external" href="https://ftp.mcs.anl.gov/pub/candle/public/improve/model_curation_data/GraphDRP/data/">this ftp location</a>.</p>
<p>CCL omics data and treatment response data (IC50 and AUC) were downloaded from the authors’ GitHub page, [GDSC_DATASET_S1-S12.zip](<a class="reference external" href="https://github.com/Jinyu2019/Suppl-data-BBpaper/blob/master/GDSC_DATASET_S1-S12.zip">https://github.com/Jinyu2019/Suppl-data-BBpaper/blob/master/GDSC_DATASET_S1-S12.zip</a>). The [protein links](<a class="reference external" href="https://stringdb-static.org/download/protein.links.v11.5/9606.protein.links.v11.5.txt.gz">https://stringdb-static.org/download/protein.links.v11.5/9606.protein.links.v11.5.txt.gz</a>) and [protein info](<a class="reference external" href="https://stringdb-static.org/download/protein.info.v11.5/9606.protein.info.v11.5.txt.gz">https://stringdb-static.org/download/protein.info.v11.5/9606.protein.info.v11.5.txt.gz</a>) files (version 11.5) for <em>Homo sapiens</em> can be obtained from the STRING website. Refer to [Data](Data.md) for more info regarding the raw data provided with the original DrugGCN model repo and preprocessing scripts allowing to generate ML data for model training and testing.</p>
<p>The raw data includes the following for the L1000 and Var1000 datasets:
1) Gene expression data. <cite>EXP.csv</cite>: Gene expression data for 734 cell lines obtained from the authors’ GitHub page, [GDSC_DATASET_S1-S12.zip](<a class="reference external" href="https://github.com/Jinyu2019/Suppl-data-BBpaper/blob/master/GDSC_DATASET_S1-S12.zip">https://github.com/Jinyu2019/Suppl-data-BBpaper/blob/master/GDSC_DATASET_S1-S12.zip</a>), and found in the file,  <cite>Table_S1_GDSC_Gene_expression.csv</cite>.
2) Protein info. <cite>PPI_INFO.txt</cite>: List of proteins including their display names and descriptions from the STRING database. This data can be downloaded from the STRING database [here](<a class="reference external" href="https://stringdb-static.org/download/protein.info.v11.5/9606.protein.info.v11.5.txt.gz">https://stringdb-static.org/download/protein.info.v11.5/9606.protein.info.v11.5.txt.gz</a>).
3) Protein links. <cite>PPI_LINK.txt</cite>: Protein network data (full network, scored links between proteins) from the STRING database. This data can be downloaded from the STRING database [here](<a class="reference external" href="https://stringdb-static.org/download/protein.links.v11.5/9606.protein.links.v11.5.txt.gz">https://stringdb-static.org/download/protein.links.v11.5/9606.protein.links.v11.5.txt.gz</a>).
4) List of genes. <cite>LIST.txt</cite>: 663 or 1000 genes for the L1000 and Var1000 datasets respectively.</p>
<p>The raw data is available in: <a class="reference external" href="https://ftp.mcs.anl.gov/pub/candle/public/improve/model_curation_data/DrugGCN/druggcn_data.tar.gz">https://ftp.mcs.anl.gov/pub/candle/public/improve/model_curation_data/DrugGCN/druggcn_data.tar.gz</a></p>
</div>
<div class="section" id="ml-data">
<h3>ML data<a class="headerlink" href="#ml-data" title="Permalink to this headline">¶</a></h3>
<p>The script <a class="reference external" href="https://github.com/JDACS4C-IMPROVE/GraphDRP/blob/develop/preprocess.py">preprocess.py</a> uses raw data to generate ML data that can be used to train and test with GraphDRP. The necessary raw data are automatically downloaded from the FTP server using a <cite>candle_lib</cite> utility function <cite>get_file()</cite> and processed:</p>
<ul class="simple">
<li><p><strong>Response data</strong>. IC50 values (PANCANCER_IC.csv) are transformed using 1 / (1 + pow(math.exp(float(ic50)), -0.1)).</p></li>
<li><p><strong>Cancer features</strong>. 735 binary features, including mutations and copy number alterations, are not modified.</p></li>
<li><p><strong>Drug features</strong>. SMILES string of each drug is converted into graph structure where nodes represent atoms and edges represent the bonds (each atom is represented by 78 features).</p></li>
</ul>
<p>The user can specify one of three data splitting strategies: 1) mixed set (random split), 2) cell-blind (hard partition on cell line samples), 3) drug-blind (hard partition on drugs).
In either case, the script generates three files, <cite>train_data.pt</cite>, <cite>val_data.pt</cite>, and <cite>test_data.pt</cite> with 0.8/0.1/0.1 ratio, and saves them in appropriate directories:</p>
<ul class="simple">
<li><p>./data_processed/&lt;split_strategy&gt;/processed/train_data.pt</p></li>
<li><p>./data_processed/&lt;split_strategy&gt;/processed/val_data.pt</p></li>
<li><p>./data_processed/&lt;split_strategy&gt;/processed/test_data.pt</p></li>
</ul>
<p>Bash script <a class="reference external" href="https://github.com/JDACS4C-IMPROVE/GraphDRP/blob/develop/preprocess_batch.sh">preprocess_batch.sh</a> generates the nine possible ML data configurations (i.e., 3 files for each splitting strategy). All the ML data files for the three splitting strategies are available in <a class="reference external" href="https://ftp.mcs.anl.gov/pub/candle/public/improve/model_curation_data/GraphDRP/data_processed/">this ftp location</a>.</p>
<p>The script <cite>dataProcess.py</cite> uses raw data to generate ML data that can be used to train and test with DrugGCN. The necessary raw data are automatically downloaded from the FTP server using the <cite>candle_lib</cite> utility function <cite>get_file()</cite> and processed.</p>
<p>The ML data files are available in FTP: <a class="reference external" href="https://ftp.mcs.anl.gov/pub/candle/public/improve/model_curation_data/DrugGCN/druggcn_data_processed.tar.gz">https://ftp.mcs.anl.gov/pub/candle/public/improve/model_curation_data/DrugGCN/druggcn_data_processed.tar.gz</a>. These files can be automatically downloaded from the FTP server using the <cite>candle_lib</cite> utility function <cite>get_file()</cite>.</p>
<p>The ML data files include the following:</p>
<ul class="simple">
<li><p>__Cancer <a href="#id1"><span class="problematic" id="id2">features__</span></a>. L1000 and Var1000 folders with gene expression and protein-protein interaction files:
- <cite>groupEXP.csv</cite>. Gene expression data of selected genes.
- <cite>groupEXP_foldChange.csv</cite>. Fold change of gene expression data of selected genes.
- <cite>groupPPI.csv</cite>. Protein-protein interaction data of selected genes where values are weights of the interactions that reflect the amount of available evidence of the interaction between two genes.</p></li>
<li><p>__Response <a href="#id1"><span class="problematic" id="id3">data__</span></a>. IC50 or AUC values of 201 drugs and 734 cell lines.
- <cite>Table_S6_GDSC_Drug_response_IC50.csv</cite>
- <cite>Table_S7_GDSC_Drug_response_AUC.csv</cite></p></li>
</ul>
<p>The user can specify which dataset, L1000 or Var1000, and which response data, IC50 or AUC, to use when training and testing the model.</p>
</div>
</div>
<div class="section" id="evaluation">
<h2>Evaluation<a class="headerlink" href="#evaluation" title="Permalink to this headline">¶</a></h2>
<p>Four evaluation schemes were used for the analysis of prediction performance.</p>
<ul class="simple">
<li><p>L1000-IC50: Gene set of 663 genes using the list of landmark genes derived from LINCS L1000 project and IC50 drug response values.</p></li>
<li><p>L1000-AUC: Gene set of 663 genes using the list of landmark genes derived from LINCS L1000 project and AUC drug response values.</p></li>
<li><p>Var1000-IC50: Gene set of the top 1000 variable genes and IC50 drug response values.</p></li>
<li><p>Var1000-AUC: Gene set of the top 1000 variable genes and AUC drug response values.</p></li>
</ul>
</div>
<div class="section" id="urls">
<h2>URLs<a class="headerlink" href="#urls" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p><a class="reference external" href="https://github.com/hauldhut/GraphDRP">Original GitHub</a></p></li>
<li><p><a class="reference external" href="https://github.com/JDACS4C-IMPROVE/GraphDRP/tree/develop">IMPROVE GitHub</a></p></li>
<li><p><a class="reference external" href="https://ftp.mcs.anl.gov/pub/candle/public/improve/model_curation_data/GraphDRP/">Data</a></p></li>
<li><p>[Original GitHub](<a class="reference external" href="https://github.com/BML-cbnu/DrugGCN">https://github.com/BML-cbnu/DrugGCN</a>)</p></li>
<li><p>[IMPROVE GitHub](<a class="reference external" href="https://github.com/JDACS4C-IMPROVE/DrugGCN/tree/develop">https://github.com/JDACS4C-IMPROVE/DrugGCN/tree/develop</a>)</p></li>
<li><p>[Data](<a class="reference external" href="https://ftp.mcs.anl.gov/pub/candle/public/improve/model_curation_data/DrugGCN/">https://ftp.mcs.anl.gov/pub/candle/public/improve/model_curation_data/DrugGCN/</a>)</p></li>
</ul>
</div>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<p>Kim S, Bae S, Piao Y, Jo K. Graph Convolutional Network for Drug Response Prediction Using Gene Expression Data. <em>Mathematics</em>. 2021; 9(7):772. <a class="reference external" href="https://doi.org/10.3390/math9070772">https://doi.org/10.3390/math9070772</a></p>
</div>
</div>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, IMPROVE Team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>