<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>NIHGCN &mdash; IMPROVE 0.1 documentation</title><link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            IMPROVE
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="quickstart.html">Quickstart Guides</a></li>
<li class="toctree-l1"><a class="reference internal" href="acknowledgment.html">Acknowledgments</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">IMPROVE</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">NIHGCN</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/NIHGCN.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="section" id="nihgcn">
<h1>NIHGCN<a class="headerlink" href="#nihgcn" title="Permalink to this headline">¶</a></h1>
<p>NIHGCN is a framework for predicting drug response using a graph neural network approach to learn cell and drug embeddings.</p>
<div class="section" id="paper">
<h2>Paper<a class="headerlink" href="#paper" title="Permalink to this headline">¶</a></h2>
<p>Peng, W., Liu, H., Dai, W., Yu, N., &amp; Wang, J. (2022). Predicting cancer drug response using parallel heterogeneous graph convolutional networks with neighborhood interactions. Bioinformatics, 38(19), 4546-4553.
<a class="reference external" href="https://doi.org/10.1093/bioinformatics/btac574">https://doi.org/10.1093/bioinformatics/btac574</a></p>
</div>
<div class="section" id="structure">
<h2>Structure<a class="headerlink" href="#structure" title="Permalink to this headline">¶</a></h2>
<p>NIHGCN is a graph neural network model that builds on a bipartite heterogeneous network based on the known drug responses in cell lines, i.e. sensitive or resistant. This bipartite graph is then used to learn the feature representation for both drugs and cell lines by transforming and aggregating the neighborhood feature information on the graph architecture using an interaction module. The interaction module consists of a Parallel Graph Convolution Network (PGCN) layer and a Neighbor Interaction (NI) layer, aggregating features from neighbors at the node and element level. Initial embeddings for cell and drug nodes are given by a vector of the normalized gene expression in a given cell line and the linear transform of the drug molecular fingerprints, respectively. The PGCN then consists of a summation of features from a cell node and all connected drug nodes multiplied by a learned weight matrix, and vice-versa for drug nodes. The NI layer takes the element-wise dot product of the PGCN summation and the node, with the goal of emphasizing similarities between vectors. Finally the aggregation layer passes a weighted sum of the PGCN and NI representations through a ReLu. Linear correlation between the aggregated drug and cell representations are used for training drug response prediction with BCE loss.</p>
</div>
<div class="section" id="data">
<h2>Data<a class="headerlink" href="#data" title="Permalink to this headline">¶</a></h2>
<div class="section" id="data-sources">
<h3>Data sources<a class="headerlink" href="#data-sources" title="Permalink to this headline">¶</a></h3>
<p>The primary data sources that have been used to construct datasets for model training and testing include</p>
<ul>
<li><p>GDSC</p>
<ul class="simple">
<li><p>The</p>
<ul>
<li><p><a class="reference external" href="https://www.cancerrxgene.org/gdsc1000/GDSC1000_WebResources//Data/suppData/TableS4A.xlsx">https://www.cancerrxgene.org/gdsc1000/GDSC1000_WebResources//Data/suppData/TableS4A.xlsx</a></p></li>
<li><p><a class="reference external" href="https://www.cancerrxgene.org/gdsc1000/GDSC1000_WebResources//Data/suppData/TableS5C.xlsx">https://www.cancerrxgene.org/gdsc1000/GDSC1000_WebResources//Data/suppData/TableS5C.xlsx</a></p></li>
</ul>
</li>
<li><p>The tables are binarized into the following tables. Files provided by the github repo for reproducibility:</p></li>
</ul>
<blockquote>
<div><ul class="simple">
<li><p>cell_drug.csv records the log IC50 association matrix of cell line-drug</p></li>
<li><p>cell_drug_binary.csv records the binary cell line-drug association matrix</p></li>
<li><p>cell_exprs.csv records cell line gene expression features</p></li>
<li><p>drug_feature.csv records the fingerprint features of drugs</p></li>
<li><p>null_mask.csv records the null values in the cell line-drug association matrix</p></li>
<li><p>threshold.csv records the drug sensitivity threshold</p></li>
</ul>
</div></blockquote>
</li>
<li><p>CCLE</p>
<ul class="simple">
<li><p>The CCLE database provides 11 670 records of cell line-drug trials. Each record reports experimental information such as drug target, dose, log(IC50) and effective area.</p></li>
<li><p>These are binarized into the following:</p>
<ul>
<li><p>cell_drug.csv records the log IC50 association matrix of cell line-drug</p></li>
<li><p>cell_drug_binary.csv records the binary cell line-drug association matrix</p></li>
<li><p>cell_exprs.csv records cell line gene expression features</p></li>
<li><p>drug_feature.csv records the fingerprint features of drugs</p></li>
</ul>
</li>
</ul>
</li>
</ul>
<p>#. PDX
#.. From the PDX Encyclopedia dataset (PDX), dataset is available in Supplementary File of Gao et al. (2015), containing gene expression profiles and drug responses values.
#.. Binarized files:
#… pdx_response.csv records the binary patient-drug association matrix.
#… pdx_exprs.csv records patient gene expression features.
#… pdx_null_mask.csv records the null values in the patient-drug association matrix.
#… drug_feature.csv records the fingerprint features of drugs.
#. TCGA
#.. The corresponding gene expression of the TCGA samples was from FirehoseBroadGDAC. TCGA drug responses were divided into two groups, responses (‘Complete Response’ and ‘Partial Response’) and non-response (‘Stable Disease’ and ‘Progressive Disease’).
#.. Binarized files:
#… patient_drug_binary.csv records the binary patient-drug association matrix.
#… tcga_exprs.csv records patient gene expression features.
#… tcga_null_mask.csv records the null values in the patient-drug association matrix.
#… drug_feature.csv records the fingerprint features of drugs.</p>
</div>
</div>
</div>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, IMPROVE Team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>