<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>PathDSP &mdash; IMPROVE 0.1 documentation</title><link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/clipboard.min.js"></script>
        <script src="../_static/copybutton.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            IMPROVE
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="quickstart.html">Quickstart Guides</a></li>
<li class="toctree-l1"><a class="reference internal" href="acknowledgment.html">Acknowledgments</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">IMPROVE</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">PathDSP</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/PathDSP.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <div class="section" id="pathdsp">
<h1>PathDSP<a class="headerlink" href="#pathdsp" title="Permalink to this headline">¶</a></h1>
<p>PathDSP is a framework for “Explainable drug sensitivity prediction through cancer pathway enrichment” (title of paper).</p>
<div class="section" id="structure">
<h2>Structure<a class="headerlink" href="#structure" title="Permalink to this headline">¶</a></h2>
<p>From Abstract: PathDSP: a pathway-based model for drug sensitivity prediction that integrates chemical structure information with enrichment of cancer signaling pathways across drug-associated genes, gene expression, mutation and copy number variation data to predict drug response on the Genomics of Drug Sensitivity in Cancer dataset. Using a deep neural network, we outperform state-of-the-art deep learning models, while demonstrating good generalizability a separate dataset of the Cancer Cell Line Encyclopedia as well as provide explainable results, demonstrated through case studies that are in line with current knowledge. Additionally, our pathway-based model achieved a good performance when predicting unseen drugs and cells, with potential utility for drug development and for guiding individualized medicine.</p>
</div>
<div class="section" id="data">
<h2>Data<a class="headerlink" href="#data" title="Permalink to this headline">¶</a></h2>
<div class="section" id="data-sources">
<h3>Data sources<a class="headerlink" href="#data-sources" title="Permalink to this headline">¶</a></h3>
<p>The primary data sources that have been used to construct datasets for model training and testing (i.e., ML data) include:</p>
<p>“Drug sensitivity data, cell-line gene expression, somatic mutation and copy number variation data for 319 cancer cell lines and 153 drugs was downloaded from Genomics of Drug Sensitivity in Cancer (GDSC downloaded from <a class="reference external" href="https://www.cancerrxgene.org/downloads/bulk_download">https://www.cancerrxgene.org/downloads/bulk_download</a>.) Drug sensitivity data of 24 drugs and 478 cancer cell lines from the Cancer Cell Line Encyclopedia (CCLE) were downloaded through the DepMap portal (<a class="reference external" href="https://depmap.org/portal/download/">https://depmap.org/portal/download/</a>, release version: public 20Q1), which also include gene expression, mutation and copy number variation data for those cancer cell lines. Primary target data was downloaded from GDSC, and PID pathways were downloaded from MSigDB. Protein–protein network was downloaded from STRING database, including protein interactions, co-expression and text-mined interactions.”</p>
</div>
<div class="section" id="raw-data">
<h3>Raw data<a class="headerlink" href="#raw-data" title="Permalink to this headline">¶</a></h3>
<p>The raw data includes the following:</p>
<ol class="arabic simple">
<li><p><cite>GDSCv2.Gao2015.Powell2020.Lee2021.GeoSearch.Ding2016.CHEM.256.MBits.txt</cite>: Looks like drug response from GDSC.</p></li>
<li><p><cite>GDSCv2.Gao2015.Powell2020.Lee2021.GeoSearch.Ding2016.DGNet.NetPEA.txt</cite>: “We collected all mutations for each cell line to perform network-based pathway enrichment analysis by the NetPEA algorithm, which calculates an enrichment score by measuring the closeness of pathway genes to a given gene set within a protein–protein interaction (PPI) network.”</p></li>
<li><p><cite>GDSCv2.Powell2020.EXP.ssGSEA.txt</cite>: “Gene expression data were measured by transcripts per million (TPM) and log-transformed. We imputed with mean for the rest of missing values. Enrichment score (ES) of each PID pathway in each cell line was calculated using the single-sample Gene Set Enrichment (ssGSEA) algorithm52 through GSEApy (<a class="reference external" href="https://gseapy.readthedocs.io/en/master/gseapy_example.html">https://gseapy.readthedocs.io/en/master/gseapy_example.html</a>). We ran permutation test for 1000 times and normalized ES scores by the size of gene set to obtain normalized ES (i.e., NES). We used the resulting pathway enrichment matrix with size of 319 cancer cell lines by 196 pathways as the gene expression feature (EXP).”</p></li>
<li><p><cite>GDSCv2.resp_PowellAUC.Alias.txt</cite>: Set of aliases [Therapy | Sample | Response | Cancer types | Datasets | Alias]</p></li>
</ol>
<p>The raw data is available in <a class="reference external" href="http://chia.team/IMPROVE_data/data">this highly suspicious location</a>.</p>
</div>
<div class="section" id="ml-data">
<h3>ML data<a class="headerlink" href="#ml-data" title="Permalink to this headline">¶</a></h3>
<p>The script <a class="reference external" href="https://github.com/JDACS4C-IMPROVE/DrugGCN/blob/develop/dataProcess.py">dataProcess.py</a> uses raw data to generate ML data that can be used to train and test with DrugGCN. The necessary raw data are automatically downloaded from the FTP server using the <cite>candle_lib</cite> utility function <cite>get_file()</cite> and processed.</p>
<p>The ML data files are available in <a class="reference external" href="https://ftp.mcs.anl.gov/pub/candle/public/improve/model_curation_data/DrugGCN/druggcn_data_processed.tar.gz">this FTP location</a>. These files can be automatically downloaded from the FTP server using the <cite>candle_lib</cite> utility function <cite>get_file()</cite>.</p>
<p>The ML data files include the following:</p>
<ul class="simple">
<li><p><strong>Cancer features</strong>.  L1000 and Var1000 folders with gene expression and protein-protein interaction files:</p>
<ul>
<li><p><cite>groupEXP.csv</cite>: Gene expression data of selected genes.</p></li>
<li><p><cite>groupEXP_foldChange.csv</cite>: Fold change of gene expression data of selected genes.</p></li>
<li><p><cite>groupPPI.csv</cite>: PPI data of selected genes where values are weights of the interactions that reflect the amount of available evidence of the interaction between two genes.</p></li>
</ul>
</li>
<li><p><strong>Response data</strong>.  IC50 or AUC values of 201 drugs and 734 cell lines.</p>
<ul>
<li><p><cite>Table_S6_GDSC_Drug_response_IC50.csv</cite></p></li>
<li><p><cite>Table_S7_GDSC_Drug_response_AUC.csv</cite></p></li>
</ul>
</li>
</ul>
<p>The user can specify which dataset, L1000 or Var1000, and which response data, IC50 or AUC, to use when training and testing the model.</p>
</div>
</div>
<div class="section" id="evaluation">
<h2>Evaluation<a class="headerlink" href="#evaluation" title="Permalink to this headline">¶</a></h2>
<p>Four evaluation schemes were used for the analysis of prediction performance.</p>
<ul class="simple">
<li><p>L1000-IC50: Gene set of 663 genes using the list of landmark genes derived from LINCS L1000 project and IC50 drug response values.</p></li>
<li><p>L1000-AUC: Gene set of 663 genes using the list of landmark genes derived from LINCS L1000 project and AUC drug response values.</p></li>
<li><p>Var1000-IC50: Gene set of the top 1000 variable genes and IC50 drug response values.</p></li>
<li><p>Var1000-AUC: Gene set of the top 1000 variable genes and AUC drug response values.</p></li>
</ul>
</div>
<div class="section" id="urls">
<h2>URLs<a class="headerlink" href="#urls" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p><a class="reference external" href="https://github.com/TangYiChing/PathDSP">Original Developer’s GitHub</a></p></li>
<li><p><a class="reference external" href="https://github.com/JDACS4C-IMPROVE/PathDSP/tree/develop">IMPROVE GitHub</a></p></li>
<li><p><a class="reference external" href="http://chia.team/IMPROVE_data/data">Data</a></p></li>
</ul>
</div>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<p>Tang, Yi-Ching, and Assaf Gottlieb. “Explainable drug sensitivity prediction through cancer pathway enrichment.” Scientific reports 11, no. 1 (2021): 1-10. <a class="reference external" href="https://www.nature.com/articles/s41598-021-82612-7">https://www.nature.com/articles/s41598-021-82612-7</a></p>
</div>
</div>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, IMPROVE Team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>